lib/application/dex_blockchain.dart

```diff
@@ -23,30 +24,26 @@ DexBlockchainsRepository _dexBlockchainsRepository(
 Future<List<DexBlockchain>> _getBlockchainsList(
   _GetBlockchainsListRef ref,
 ) async {
-  final blockchainsList = await ref
-      .watch(_dexBlockchainsRepositoryProvider)
-      .getBlockchainsListConf();
+  final blockchainsList = await ref.watch(_getBlockchainsListProvider.future);
   return ref
       .watch(_dexBlockchainsRepositoryProvider)
       .getBlockchainsList(blockchainsList);
 }
 
 @riverpod
-Future<DexBlockchain?> _getBlockchainFromEnv(
-  _GetBlockchainFromEnvRef ref,
-  String env,
+Future<DexBlockchain?> _currentBlockchain(
+  _CurrentBlockchainRef ref,
 ) async {
-  final blockchainsList = await ref
-      .watch(_dexBlockchainsRepositoryProvider)
-      .getBlockchainsListConf();
+  final blockchainsList = await ref.watch(_getBlockchainsListProvider.future);
 
+  final environment = ref.watch(environmentProvider);
   return ref
       .watch(_dexBlockchainsRepositoryProvider)
-      .getBlockchainFromEnv(blockchainsList, env);
+      .getBlockchainFromEnv(blockchainsList, environment.name);
 }
 
 abstract class DexBlockchainsProviders {
```

Comment:
Doesn't keeping this in the provider pose a problem for provider generation?

Reply:
I'm not sure to understand what is wrong here.